### PR TITLE
Nexus-36690 azure ha preview helm

### PR DESCRIPTION
**Pull request by bobotimi**
Nexus-36690

### CODE REVIEW

File: nxrm-azure-ha-helm/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+    of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+    Eclipse Foundation. All other trademarks are the property of their respective owners.
+
+-->
+
+# Helm Chart for a High-Availability Nexus Repository Deployment in AWS
+
+This Helm chart configures the Kubernetes resources that are needed for a high-availability (HA) Nexus Repository deployment on AWS.
```
Comment (oleksiirudyk) [inline nxrm-azure-ha-helm/README.md:18]: .. in Azure.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+/.idea/
+.DS_STORE
```
Comment (oleksiirudyk) [inline .gitignore:2]: nit: add new line

File: nxrm-azure-ha-helm/templates/nxrm-logback-tasklogfile-override.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        <Append>true</Append>
+        <encoder class="org.sonatype.nexus.pax.logging.NexusLayoutEncoder">
+          <pattern>%d{"yyyy-MM-dd HH:mm:ss,SSSZ"} %-5p [%thread] %node %mdc{userId:-*SYSTEM} %c - %m%n</pattern>
+        </encoder>
+        <rollingPolicy class="ch.qos.logback.core.rolling.TimeBasedRollingPolicy">
+          <fileNamePattern>${karaf.data}/log/tasks/allTasks-%d{yyyy-MM-dd}.log.gz</fileNamePattern>
+          <maxHistory>1</maxHistory>
+        </rollingPolicy>
+      </appender>
+    </included>
```
Comment (oleksiirudyk) [inline nxrm-azure-ha-helm/templates/nxrm-logback-tasklogfile-override.yaml:21]: nit:new line

File: nxrm-azure-ha-helm/templates/statefulset.yaml [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+                path: logback-tasklogfile-appender-override.xml
+  volumeClaimTemplates:
+    - metadata:
+        name: nexus-data
+      spec:
+        accessModes: [ "{{.Values.pvc.accessModes }}" ]
+        storageClassName: "{{.Values.storageClass.name}}"
+        resources:
+          requests:
+            storage: {{.Values.pvc.storage }}
```
Comment (oleksiirudyk) [inline nxrm-azure-ha-helm/templates/statefulset.yaml:128]: nit:new line

File: nxrm-azure-ha-helm/README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+---
+## Installing this Chart
+
+1. Check out this git repository.
+
+2. Install this chart using the following:
+  
+```helm install nxrm nxrm3-preview-ha-repository/nxrm-aws-ha-helm -f values.yaml```
```
Comment (oleksiirudyk) [inline nxrm-azure-ha-helm/README.md:27]: nxrm-azure-ha-helm

File: nxrm-azure-ha-helm/Chart.yaml [hunk 6]
```diff
@@ -0,0 +1,41 @@
+apiVersion: v2
```
Comment (Mykyta) [inline nxrm-azure-ha-helm/Chart.yaml:1]: missed license header

File: nxrm-azure-ha-helm/templates/statefulset.yaml [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+              valueFrom:
+                secretKeyRef:
+                  name: nxrm-admin-secret
+                  key: nexus-admin-password
+            - name: NEXUS_SECURITY_RANDOMPASSWORD
+              value: "false"
+            - name: INSTALL4J_ADD_VM_PARAMS
+              value: "{{ .Values.statefulset.container.env.install4jAddVmParams }} -Dnexus.licenseFile=/nxrm-secrets/{{ .Values.secret.license.name }} \
+          -Dnexus.datastore.clustered.enabled=true -Djava.util.prefs.userRoot=${NEXUS_DATA}/javaprefs \
+          -Dnexus.datastore.nexus.jdbcUrl=jdbc:postgresql://${DB_HOST}:{{ .Values.statefulset.container.env.nexusDBPort }}/${DB_NAME}?sslmode=require \
```
Comment (Mykyta) [inline nxrm-azure-ha-helm/templates/statefulset.yaml:76]: change the `{{ .Values.statefulset.container.env.nexusDBPort }}` to `${DB_PORT}`

File: nxrm-azure-ha-helm/values.yaml [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+  name: nxrm-statefulset
+  clusterName: nxrm-nexus
+  initContainer:
+    image:
+      repository: busybox
+      tag: 1.33.1
+  container:
+    image:
+      repository: sonatype/nexus3
+      tag: 3.44.0
```
Comment (Mykyta) [inline nxrm-azure-ha-helm/values.yaml:15]: 3.45.1

**Review by Mykyta — COMMENTED** [review]
Add license headers to all yaml files

File: nxrm-azure-ha-helm/Chart.yaml (hunk 6 above)
Comment (bobotimi) [reply nxrm-azure-ha-helm/Chart.yaml:1]: I don't see license in the AWS ones. What've we've done here is to include a [LICENSE](https://github.com/sonatype/nxrm3-preview-ha-repository/blob/main/nxrm-aws-ha-helm/LICENSE) file

**Comment by bobotimi** [comment]
> Add license headers to all yaml files

Already includes [LICENSE file](https://github.com/sonatype/nxrm3-preview-ha-repository/blob/dfa873f9fd9b6e26e80db31332f8169f1d912704/nxrm-azure-ha-helm/LICENSE)

File: nxrm-azure-ha-helm/Chart.yaml (hunk 6 above)
Comment (Mykyta) [reply nxrm-azure-ha-helm/Chart.yaml:1]: This one https://github.com/sonatype/nxrm3-preview-ha-repository/blob/main/sample-aws-ha-yamls/aws-ha-external-dns-rbac.yaml for example. As far as I know, each file should contain license header

File: nxrm-azure-ha-helm/templates/statefulset.yaml (hunk 7 above)
Comment (Mykyta) [reply nxrm-azure-ha-helm/templates/statefulset.yaml:76]: Yes, but we changed it for our resiliency yamls, see https://github.com/sonatype/nxrm-sample-files-repo/blob/main/aws-resiliency-yamls/aws-resiliency-nxrm-deployment.yaml#L78

**Review by oleksiirudyk — APPROVED** [review]

**Review by Mykyta — APPROVED** [review]

**Pull request merged**